spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/support/SimpleSpannerRepository.java

```diff
@@ -154,8 +157,36 @@ public Page findAll(Pageable pageable) {
 			}
 		}
 		else {
-			k = key instanceof Key ? (Key) key : Key.of(key);
+			k = Key.class.isAssignableFrom(key.getClass()) ? (Key) key
+					: Key.of(convertKeyPart(key));
 		}
 		return operation.apply(k);
 	}
+
+	private Object convertKeyPart(Object object) {
+		if (MappingSpannerConverter
+				.isValidSpannerKeyType(ConversionUtils.boxIfNeeded(object.getClass()))) {
+			return object;
+		}
+		/*
+		 * Iterate through the supported Key component types in the same order as the
+		 * write converter. For example, if a type can be converted to both String and
+		 * Double, we want both the this key conversion and the write converter to choose
+		 * the same.
+		 */
+		for (Class validKeyType : MappingSpannerWriteConverter.singleItemType2ToMethodMap
```

Comment:
same thing here - I would rather hide the detail of `singleItemType2ToMethodMap.keySet()` and expose a non-static member function `writableTypes()`

Reply:
changes made